sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisCursor.java

```diff
@@ -0,0 +1,144 @@
+
+  public static RedisCursor of(String cursor, long dbSize, boolean isStart) {
+    return new RedisCursor(cursor, dbSize, isStart);
+  }
+
+  private RedisCursor(String cursor, long dbSize, boolean isStart) {
+    this.cursor = cursor;
+    this.dbSize = dbSize;
+    this.isStart = isStart;
+  }
+
+  /**
+   * {@link RedisCursor} implements {@link Comparable Comparable&lt;RedisCursor&gt;} by transforming
+   * the cursors to an index of the Redis table.
+   */
+  @Override
+  public int compareTo(@Nonnull RedisCursor other) {
+    checkNotNull(other, "other");
+    return Long.compare(Long.parseLong(cursor), Long.parseLong(other.cursor));
+  }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    RedisCursor that = (RedisCursor) o;
+    return dbSize == that.dbSize && isStart == that.isStart && Objects.equals(cursor, that.cursor);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(cursor, dbSize, isStart);
+  }
+
+  public String getCursor() {
+    return cursor;
+  }
+
+  public long getDbSize() {
+    return dbSize;
+  }
+
+  @VisibleForTesting
+  static ByteKey redisCursorToByteKey(RedisCursor cursor) {
+    if ("0".equals(cursor.getCursor())) {
+      if (cursor.isStart) {
+        return ByteKey.of(0x00);
+      } else {
+        return ByteKey.EMPTY;
+      }
+    }
+    int nBits = getTablePow(cursor.dbSize);
+    long cursorLong = Long.parseLong(cursor.getCursor());
+    long reversed = shiftBits(cursorLong, nBits);
+    BigEndianLongCoder coder = BigEndianLongCoder.of();
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    try {
+      coder.encode(reversed, os);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("invalid redis cursor " + cursor);
+    }
+    byte[] byteArray = os.toByteArray();
+    return ByteKey.copyFrom(byteArray);
+  }
+
+  @VisibleForTesting
+  static long shiftBits(long a, int nBits) {
+    long b = 0;
+    for (int i = 0; i < nBits; ++i) {
+      b <<= 1;
+      b |= (a & 1);
+      a >>= 1;
+    }
+    return b;
+  }
+
+  @VisibleForTesting
+  static int getTablePow(long nKeys) {
+    return 64 - Long.numberOfLeadingZeros(nKeys - 1);
+  }
+
+  @VisibleForTesting
+  static RedisCursor byteKeyToRedisCursor(ByteKey byteKeyStart, long nKeys, boolean isStart) {
+    if (byteKeyStart.isEmpty() || byteKeyStart.equals(ZERO_KEY)) {
+      return RedisCursor.of("0", nKeys, isStart);
+    }
+    int nBits = getTablePow(nKeys);
+    ByteBuffer bb = ByteBuffer.wrap(byteKeyStart.getBytes());
+    if (bb.capacity() < nBits) {
+      int rem = nBits - bb.capacity();
+      byte[] padding = new byte[rem];
+      bb = ByteBuffer.allocate(nBits).put(padding).put(bb.array());
+      bb.position(0);
+    }
+    long l = bb.getLong();
+    l = shiftBits(l, nBits);
+    return RedisCursor.of(Long.toString(l), nKeys, isStart);
+  }
```

Comment:
It makes sense to have these methods within the RedisCursorRangeTracker since that is where you need to perform the conversions.